### ISSUE TITLE
PDF Export failing to export at right size

### ISSUE THREAD

**Issue by StoneOfLight** [lifelike/countersheetsextension#17]
When exporting to PDF, the content of the page is shrunk to about 20% of its original size. The svg was created in Inkscape 0.92 and I'm using the latest version of the extension.

[pdftest.zip](https://github.com/lifelike/countersheetsextension/files/1188319/pdftest.zip)

**Comment by lifelike** [comment]
The PDF export function works by calling inkscape with some command-line arguments. With log enabled when running the extension you can see exactly what arguments are used. I copied that line and experimented a bit on the command-line and as expected gets the same error.

The -i flag is used to render PDF for only one layer (at a time). For instance -i cs_layer_1 to render the PDF for the first layer. If that flag is removed the PDF gets the correct size (but then, of course, every PDF contains all layers, so they will all look the same and the PDF export is rather pointless...).

Edited the generated countersheet SVG and made a very minimalistic version with only the cs_layer_1 group containing an empty rectangle. It still shows the same error. It must either be a bug in inkscape or some clever feature that looks like a bug to me.

Strange thing is I do not see this on my other files. There must be something special about some attributes in this SVG that triggers this behavior.

**Comment by lifelike** [comment]
Based on that pdftest.svg and running the extension, then removing a lot of stuff from the SVG, I came up with this minimal test:
https://gist.github.com/lifelike/1ad9088f2590d151a0b10558870dc500

Reporting this as a bug for the Inkscape project.

**Comment by lifelike** [comment]
This seems to be a problem in inkscape. It would be possible to work around it by generating the temporary SVG files for PDF-export with the other layers removed instead of relying on doing PDF-export for a particular layer (id).

**Comment by lifelike** [comment]
It is quite a big rewrite unfortunately to work around this by doing PDF export with one layer at a time generated, and that would also slow down PDF exports for all documents to fix this probably uncommon issue, so not planning on fixing this (merge-requests welcome!).

Here is the Inkscape bug report:
https://bugs.launchpad.net/inkscape/+bug/1714365

There is this comment there that might be a hint how to work around it by fixing the SVG document instead:
"Seems to be related to document scale. When I set scale to 3.77953 (with display units mm), export is as expected."

Perhaps by playing around a bit in document settings and/or copying the content to a new document it is possible to avoid this issue?

**Comment by lifelike** [comment]
Reconsidered because of new information in the Inkscape bug. This problem has existed in Inkscape for a long time. Besides even if it is fixed in future versions of Inkscape it would be nice to still support current versions.

**Comment by lifelike** [comment]
Added a workaround. PDFs are exported without using the -i flag to select what layer to export. Instead the SVG is saved several times with only one layer visible each time and the entire SVG is exported to PDF.

For the included test-cases at least the I did not notice any performance degradation, but it is possible for some very large generated files with many layers it will be noticeably slower.

Release (tag) 2.0.1 contains this fix.
